app/javascript/app/components/data-explorer-content/data-explorer-filters/data-explorer-filters.js

```diff
@@ -87,20 +87,11 @@ class DataExplorerContentContainer extends PureComponent {
     const removing = oldFilters && value.length < oldFilters.length;
     const selectedFilter = !oldFilters
       ? value[0]
-      : value
-        .filter(x => oldFilters.indexOf(x) === -1)
-        .concat(oldFilters.filter(x => value.indexOf(x) === -1))[0];
+      : value.filter(f => !oldFilters.map(o => o.label).includes(f.label));
```

Comment:
This logic should be changed, now it is not matching any element inside of `value` object when removing a selection.